lib/contracts/genesisProtocol.js

```diff
@@ -0,0 +1,671 @@
+    if (!options.executable) {
+      throw new Error("executable is not defined");
+    }
+
+    if (!options.proposer) {
+      throw new Error("proposer is not defined");
+    }
+
+    if ((options.numOfChoices < 1) || (options.numOfChoices > 10)) {
+      throw new Error("numOfChoices must be between 1 and 10");
+    }
+
+    const tx = await this.contract.propose(
+      options.numOfChoices,
+      options.paramsHash,
+      options.avatar,
+      options.executable,
+      options.proposer
+    );
+
+    return new ArcTransactionProposalResult(tx);
+  }
+
+  async stake(opts = {}) {
+
+    const defaults = {
+      proposalId: undefined,
+      vote: undefined,
+      amount: undefined
+    };
+
+    const options = dopts(opts, defaults, { allowUnknown: true });
+
+    if (!options.proposalId) {
+      throw new Error("proposalId is not defined");
+    }
+
+    this._validateVote(options.vote, options.proposalId);
+
+    const amount = web3.toBigNumber(options.amount);
+
+    if (amount <= 0) {
+      throw new Error("amount must be > 0");
+    }
+
+    const tx = await this.contract.stake(
+      options.proposalId
+      , options.vote
+      , amount
+    );
+
+    return new ArcTransactionResult(tx);
+  }
+
+  async vote(opts = {}) {
+
+    const defaults = {
+      proposalId: undefined,
+      vote: undefined
+    };
+
+    const options = dopts(opts, defaults, { allowUnknown: true });
+
+    if (!options.proposalId) {
+      throw new Error("proposalId is not defined");
+    }
+
+    this._validateVote(options.vote, options.proposalId);
+
+    const tx = await this.contract.vote(
+      options.proposalId
+      , options.vote
+    );
+
+    return new ArcTransactionResult(tx);
+  }
+
+  async voteWithSpecifiedAmounts(opts = {}) {
+    const defaults = {
+      proposalId: undefined,
+      vote: undefined,
+      reputation: undefined
+    };
+
+    const options = dopts(opts, defaults, { allowUnknown: true });
+
+    if (!options.proposalId) {
+      throw new Error("proposalId is not defined");
+    }
+
+    this._validateVote(options.vote, options.proposalId);
+
+    if (!options.reputation) {
+      throw new Error("reputation is not defined");
+    }
+
+    const tx = await this.contract.voteWithSpecifiedAmounts(
+      options.proposalId
+      , options.vote
+      , options.reputation
+      , 0
+    );
+
+    return new ArcTransactionResult(tx);
+  }
+
+  async redeem(opts = {}) {
+
+    const defaults = {
+      proposalId: undefined,
+      beneficiary: undefined
+    };
+
+    const options = dopts(opts, defaults, { allowUnknown: true });
+
+    if (!options.proposalId) {
+      throw new Error("proposalId is not defined");
+    }
+
+    if (!options.beneficiary) {
+      throw new Error("beneficiary is not defined");
+    }
+
+    const tx = await this.contract.redeem(
+      options.proposalId
+      , options.beneficiary
+    );
+
+    return new ArcTransactionResult(tx);
+  }
+
+  async getShouldBoost(opts = {}) {
```

Comment:
for boolean getters I personally prefer to not use get and have it just be shouldBoost, but thats just personal preference